src/Quest.ts

```diff
@@ -123,14 +123,20 @@ export class Quest extends EventEmitter {
 
 	addGoal(goal: QuestGoal) {
 		this.goals.push(goal);
-		goal.on('progress', () => this.onProgressUpdated());
+		goal.on('progress', () => this.onProgressUpdated(goal));
 	}
 
 	/**
 	 * @fires Quest#turn-in-ready
 	 * @fires Quest#progress
 	 */
-	onProgressUpdated(): void {
+	onProgressUpdated(goal: QuestGoal): void {
+		const goalProgress = goal.getProgress();
+		if (goalProgress.percent >= 100) {
```

Comment:
Consider check against peer completion here to short circuit as well. Need logs to see where double-completion is happening.